python/lsst/pipe/tasks/visualizeVisit.py

```diff
@@ -0,0 +1,250 @@
+        """
+        class ImageSource:
+            """Source of images for makeImageFromCamera"""
+            def __init__(self, exposures):
+                self.isTrimmed = True
+                self.exposures = exposures
+                self.background = np.nan
+
+            def getCcdImage(self, detector, imageFactory, binSize):
+                """Provide image of CCD to makeImageFromCamera
+
+                Parameters
+                ----------
+                detector : `int`
+                    Detector ID to get image data for.
+                imageFactory : `lsst.afw.image.Image`
+                    Type of image to construct.
+                binSize : `int`
+                    Binsize to use to recompute dimensions.
+
+                Returns
+                -------
+                image : `lsst.afw.image.Image`
+                    Appropriately rotated, binned, and transformed
+                    image to be mosaicked.
+                detector : `lsst.afw.cameraGeom.Detector`
+                    Camera detector that the returned image data
+                    belongs to.
+                """
+                detId = detector.getId()
+                if detId not in self.exposures:
+                    dims = detector.getBBox().getDimensions()/binSize
+                    image = imageFactory(*[int(xx) for xx in dims])
+                    image.set(self.background)
+                else:
+                    image = self.exposures[detector.getId()]
+                if hasattr(image, "getMaskedImage"):
+                    image = image.getMaskedImage()
+                if hasattr(image, "getMask"):
+                    mask = image.getMask()
+                    isBad = mask.getArray() & mask.getPlaneBitMask("NO_DATA") > 0
+                    image = image.clone()
+                    image.getImage().getArray()[isBad] = self.background
+                if hasattr(image, "getImage"):
+                    image = image.getImage()
+
+                image = afwMath.rotateImageBy90(image, detector.getOrientation().getNQuarter())
+                return image, detector
+
+        image = afwUtils.makeImageFromCamera(
+            camera,
+            imageSource=ImageSource(inputExps),
+            imageFactory=afwImage.ImageF,
+            binSize=binning
+        )
+        return image
+
+    def run(self, inputExps, camera):
+        """Mosaic inputs.
+
+        Parameters
+        ----------
+        inputExp : `list` [`lsst.afw.image.Exposure`]
+            Input exposure data to bin.
+        camera : `lsst.afw.cameraGeom.Camera`
+            Input camera to use for mosaic geometry.
+
+        Returns
+        -------
+        output : `lsst.pipe.base.Struct`
+            Results struct with attribute:
+
+            ``outputExp``
+                Binned version of input image. (`lsst.afw.image.Exposure`)
+        """
+        expDict = dict((exp.getDetector().getId(), exp) for exp in inputExps)
```

Comment:
This works, but is doing more work under the hood than is needed, just use dict comprehension:
```
expDict = {exp.getDetector().getId(): exp for exp in inputExps}
```